.github/workflows/companion-deploy.yml

```
@@ -2,10 +2,30 @@ name: Companion Deploy
 
 on:
   push:
-    tags:
-      - '@uppy/companion@*'
+    branches: [ 'main' ]
+    paths:
+      - yarn.lock
+      - 'packages/@uppy/companion/**'
+      - '.github/workflows/companion.yml'
 
 jobs:
+  npm:
+    name: Upload npm tarball
```

```suggestion
    name: Create npm tarball artifact
```
as far as I understand, we're not uploading it to npm, just attaching it as an artifact to the github action job (which will expire after a few weeks AFAK)